src/test/java/software/amazon/event/kafkaconnector/EventBridgeSinkConnectorS3IT.java

```diff
@@ -0,0 +1,121 @@
+    log.info("creating kafka producer");
+
+    var mapper = new ObjectMapper();
+
+    var sendTimestamp = LocalDateTime.now().toString();
+    var jsonTestEvent =
+        mapper
+            .createObjectNode()
+            .put("sentTimestamp", sendTimestamp)
+            .put("message", "hello from kafka");
+
+    // use as marker object
+    putS3Object("START", RequestBody.empty());
+
+    log.info("sending kafka json test record {} to topic {}", jsonTestEvent, TEST_RESOURCE_NAME);
+    sendKafkaRecord(new ProducerRecord<>(TEST_RESOURCE_NAME, TEST_EVENT_KEY, jsonTestEvent));
+
+    log.info("polling sqs queue {} for json test record", getQueueArn());
+    var s3ObjectArn = new AtomicReference<String>(null);
+    await()
+        .atMost(5, SECONDS)
+        .pollInterval(1, SECONDS)
+        .untilAsserted(
+            () -> {
+              var messages = receiveSqsMessages(1).messages();
+              log.info("received sqs message(s): {}", messages);
+
+              assertThat(messages)
+                  .hasSize(1)
+                  .extracting(Message::body)
+                  .satisfies(
+                      body ->
+                          MatcherAssert.assertThat(
+                              body,
+                              isJson(
+                                  withJsonPath(
+                                      "$.detail",
+                                      allOf(
+                                          hasJsonPath("partition", greaterThanOrEqualTo(0)),
+                                          hasJsonPath("offset", greaterThanOrEqualTo(0)),
+                                          hasJsonPath("topic", equalTo(TEST_EVENT_KEY)),
+                                          hasJsonPath("key", equalTo(TEST_EVENT_KEY)),
+                                          hasJsonPath(
+                                              "value.sentTimestamp", equalTo(sendTimestamp)),
+                                          hasNoJsonPath("value.message"),
+                                          hasJsonPath(
+                                              "dataref", startsWith("arn:aws:s3:::test-bucket/")),
+                                          hasJsonPath(
+                                              "datarefPath", equalTo("$.detail.value.message")))))),
+                      atIndex(0));
+
+              var document = JsonPath.parse(messages.get(0).body());
+              var dataRef = (String) document.read(JsonPath.compile("$.detail.dataref"));
+              s3ObjectArn.set(dataRef);
+            });
+
+    // use as marker object
+    putS3Object("STOP", RequestBody.empty());
+
+    var s3ObjectKey = s3ObjectArn.get().substring("arn:aws:s3:::test-bucket/".length());
+
+    var s3Objects = listS3Objects().contents();
+    log.info("received s3 object(s): {}", s3Objects);
+
+    assertThat(s3Objects)
+        .extracting(S3Object::key)
+        .containsExactlyInAnyOrder("START", "STOP", s3ObjectKey);
```

Comment:
IMHO order is wrong here, STOP should be last

Reply:
And I also know why tests are passing: `*S3` integration tests are not running. This is due to the `includes` convention in the `maven-failsafe` plugin: https://maven.apache.org/surefire/maven-failsafe-plugin/integration-test-mojo.html#includes

Reply:
I've added an assertion temporarily that waits for two S3 objects ('START' and UUID) but the order is not preserved when receiving all S3 objects:

```
received s3 object(s): [S3Object(Key=6d0e3797-e10b-4b46-b1f0-56f542385f01, LastModified=2024-05-01T09:16:21Z, ETag="e4ee9d56f4845791c0cdca
ca7dc4a200", Size=16, StorageClass=STANDARD, Owner=Owner(DisplayName=webfile, ID=75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a)), S3Object(Key=START, LastModified=2024-05-01T09:16:20Z, ETag="d41d8cd98f00b204e9800998ecf8
427e", Size=0, StorageClass=STANDARD, Owner=Owner(DisplayName=webfile, ID=75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a)), S3Object(Key=STOP, LastModified=2024-05-01T09:16:22Z, ETag="d41d8cd98f00b204e9800998ecf8427e", S
ize=0, StorageClass=STANDARD, Owner=Owner(DisplayName=webfile, ID=75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a))]
```

It looks like the key is sorted. I guess it's an implementation detail of LocalStack.

Reply:
The assertion does not take order into account.